x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/RestEsqlTestCase.java

```diff
@@ -295,6 +320,81 @@ public void testCSVNoHeaderMode() throws IOException {
     }
 
+    public void testOutOfRangeComparisons() throws IOException {
+        final int NUM_SINGLE_VALUE_ROWS = 100;
+        bulkLoadTestData(NUM_SINGLE_VALUE_ROWS);
+        bulkLoadTestData(10, NUM_SINGLE_VALUE_ROWS, false, RestEsqlTestCase::createDocumentWithMVs);
+        bulkLoadTestData(5, NUM_SINGLE_VALUE_ROWS + 10, false, RestEsqlTestCase::createDocumentWithNulls);
+
+        List<String> dataTypes = List.of(
+            "alias_integer",
+            "byte",
+            "short",
+            "integer",
+            "long",
+            // TODO: https://github.com/elastic/elasticsearch/issues/102935
+            // "unsigned_long",
+            // TODO: https://github.com/elastic/elasticsearch/issues/100130
+            // "half_float",
+            // "float",
+            "double",
+            "scaled_float"
+        );
+
+        String lessOrLessEqual = randomBoolean() ? " < " : " <= ";
+        String largerOrLargerEqual = randomBoolean() ? " > " : " >= ";
+        String inEqualPlusMinus = randomBoolean() ? " != " : " != -";
+        String equalPlusMinus = randomBoolean() ? " == " : " == -";
+        // TODO: once we do not support infinity and NaN anymore, remove INFINITY/NAN cases.
+        // https://github.com/elastic/elasticsearch/issues/98698#issuecomment-1847423390
+        String humongousPositiveLiteral = randomBoolean() ? HUMONGOUS_DOUBLE : INFINITY;
+        String nanOrNull = randomBoolean() ? NAN : "to_double(null)";
+
+        List<String> trueForSingleValuesPredicates = List.of(
+            lessOrLessEqual + humongousPositiveLiteral,
+            largerOrLargerEqual + " -" + humongousPositiveLiteral,
+            inEqualPlusMinus + humongousPositiveLiteral,
+            inEqualPlusMinus + NAN
+        );
+        List<String> alwaysFalsePredicates = List.of(
+            lessOrLessEqual + " -" + humongousPositiveLiteral,
+            largerOrLargerEqual + humongousPositiveLiteral,
+            equalPlusMinus + humongousPositiveLiteral,
+            lessOrLessEqual + nanOrNull,
+            largerOrLargerEqual + nanOrNull,
+            equalPlusMinus + nanOrNull,
+            inEqualPlusMinus + "to_double(null)"
+        );
+
+        for (String fieldWithType : dataTypes) {
+            for (String truePredicate : trueForSingleValuesPredicates) {
+                String comparison = fieldWithType + truePredicate;
+                var query = builder().query(LoggerMessageFormat.format(null, "from {} | where {}", testIndexName(), comparison));
+                List<String> expectedWarnings = List.of(
+                    "Line 1:29: evaluation of [" + comparison + "] failed, treating result as null. Only first 20 failures recorded.",
+                    "Line 1:29: java.lang.IllegalArgumentException: single-value function encountered multi-value"
+                );
+                var result = runEsql(query, expectedWarnings, mode);
+
+                var values = (ArrayList) result.get("values");
```

Comment:
Instead of doing casting import the as method from EsqTestUtils:
`var values = as(result.get("values"), List.class)`

Reply:
Will do! EsqlUtils requires to re-export esql in Gradle, otherwise the class won't load - but that's a small change.